zuul-core/src/main/java/com/netflix/zuul/netty/insights/PassportLoggingHandler.java

```diff
@@ -99,21 +99,31 @@ private void logPassport(Channel channel)
                     + ", toplevelid = " + topLevelRequestId
                     + ", req = " + request.getInfoForLogging()
                     + ", passport = " + String.valueOf(passport));
-        }
 
-        // Some logging of session states if certain criteria match:
-        if (LOG.isInfoEnabled()) {
-            if (passport.wasProxyAttempt()) {
+            StartAndEnd inReqToOutResp = passport.findFirstStartAndLastEndStates(PassportState.IN_REQ_HEADERS_RECEIVED, PassportState.OUT_REQ_LAST_CONTENT_SENT);
+            if (passport.calculateTimeBetween(inReqToOutResp) > WARN_REQ_PROCESSING_TIME_NS.get()) {
+                LOG.debug("Request processing took longer than threshold! toplevelid = " + topLevelRequestId + ", "
```

Comment:
If we actually want to improve allocation, avoiding string concatenation is a much better low hanging fruit here imo.

Reply:
We want to primarily reduce logging noise, and secondarily improve allocation.

Reply:
The PR description calls out that this is for the purpose of reducing allocation.

Reply:
what would be a better alternative to not doing String concatenation, using a ```StringBuilder``` or ```StringBuffer``` to build the string only once and then ```LOG.debug``` once?